src/search/algorithms.clj

```diff
@@ -54,3 +71,50 @@
       (= current-state start-state) (reverse (conj path start-state))
       :else (recur (came-from current-state)
                    (conj path current-state)))))
+
+(defn a-star[children-fn heuristic-fn max-states start-state goal-state]
+
+  (loop [max-states max-states ;;Generational loop
+         frontier (pm/priority-map start-state 0)
+         came-from {start-state nil}
+         cost-so-far {start-state 0}]
+    (if (or (empty? frontier)
+            (<= max-states 0)
+            (= (first (peek frontier)) goal-state))
+      came-from
+      (let [current (first (peek frontier))
+            current-cost (cost-so-far current)
+            current-children (children-fn current)
+            child-cost (+ current-cost 1) ;; increasing cost by one because all costs are the same for sudoku
```

Comment:
It is true that in *this case* (sudoku) you can just add one, but we'd want the implementation of `a-star` to be more general than that, right? Why unnecessarily bake in assumptions about sudoku?